### PR TITLE
[ET-1369] Checkout Toggle: Some JS Tweaks

### PR DESCRIPTION
**Pull request by codingmusician**
### 🎫 Ticket

[ET-1369] 

### 🗒️ Description

Made some changes for accessibility and removed the animation.

### 🎥 Artifacts 

https://i.imgur.com/jK5ea2A.gif

### ✔️ Checklist
- [ ] I've included a changelog entry in the readme.txt file. 
- [x] My code is tested. 
- [x] My code has [proper inline documentation](https://the-events-calendar.github.io/products-engineering/docs/code-standards/).

[ET-1369]: https://theeventscalendar.atlassian.net/browse/ET-1369?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/resources/js/commerce/gateway/toggler.js [hunk 1]
```diff
@@ -122,7 +113,7 @@ tribe.tickets.commerce.gateway = tribe.tickets.commerce.gateway || {};
 	 * @param {Element} gateway Gateway element to show.
 	 */
 	obj.toggler.showGateway = gateway => {
-		$( gateway ).show( obj.toggler.toggleDuration );
+		$( gateway ).show().attr('aria-expanded', 'true');
```
Comment (bordoni) [inline src/resources/js/commerce/gateway/toggler.js:116]:
```suggestion
		$( gateway ).show().attr( 'aria-expanded', 'true' );
```

File: src/resources/js/commerce/gateway/toggler.js [hunk 2]
```diff
@@ -133,7 +124,7 @@ tribe.tickets.commerce.gateway = tribe.tickets.commerce.gateway || {};
 	 * @param {Element} gateway Gateway element to hide.
 	 */
 	obj.toggler.hideGateway = gateway => {
-		$( gateway ).hide( obj.toggler.toggleDuration );
+		$( gateway ).hide().attr('aria-expanded', 'false');
```
Comment (bordoni) [inline src/resources/js/commerce/gateway/toggler.js:127]:
```suggestion
		$( gateway ).hide().attr( 'aria-expanded', 'false' );
```

File: src/resources/js/commerce/gateway/toggler.js [hunk 3]
```diff
@@ -167,7 +158,7 @@ tribe.tickets.commerce.gateway = tribe.tickets.commerce.gateway || {};
 			obj.toggler.showGateway( gateway );
 		});
 		obj.toggler.toggles.additional.addClass(obj.toggler.classes.toggleOpen);
-		obj.toggler.toggles.default.removeClass(obj.toggler.classes.toggleOpen).show();
+		obj.toggler.toggles.default.removeClass(obj.toggler.classes.toggleOpen).show().attr('aria-hidden', 'false');
```
Comment (bordoni) [inline src/resources/js/commerce/gateway/toggler.js:161]:
```suggestion
		obj.toggler.toggles.default.removeClass(obj.toggler.classes.toggleOpen).show().attr( 'aria-hidden', 'false' );
```

File: src/resources/js/commerce/gateway/toggler.js [hunk 4]
```diff
@@ -149,7 +140,7 @@ tribe.tickets.commerce.gateway = tribe.tickets.commerce.gateway || {};
 			}
 			obj.toggler.hideGateway( gateway );
 		});
-		obj.toggler.toggles.default.addClass(obj.toggler.classes.toggleOpen).hide();
+		obj.toggler.toggles.default.addClass(obj.toggler.classes.toggleOpen).hide().attr('aria-hidden', 'true');
```
Comment (bordoni) [inline src/resources/js/commerce/gateway/toggler.js:143]:
```suggestion
		obj.toggler.toggles.default.addClass(obj.toggler.classes.toggleOpen).hide().attr( 'aria-hidden', 'true' );
```

**Review by bordoni — CHANGES_REQUESTED** [review]

**Review by juanfra — COMMENTED** [review]
Looking good! I'd try to also use the `aria-selected` on the "toggler buttons".

I'd also try to toggle the a11y classes we have (`.tribe-common-a11y-hidden`), apart from using the show/hide. Because, with how things are now, the container can be not visible in the browser but it's still visible for screen readers.

**Review by bordoni — CHANGES_REQUESTED** [review]
Spacing is missing spaces on `()`

File: src/resources/js/commerce/gateway/toggler.js [hunk 5]
```diff
@@ -72,6 +72,7 @@ tribe.tickets.commerce.gateway = tribe.tickets.commerce.gateway || {};
 		toggle: 'tribe-tickets__commerce-checkout-gateway-toggle',
 		toggleOpen: 'tribe-tickets__commerce-checkout-gateway-toggle--open',
 		toggleButton: 'tribe-tickets__commerce-checkout-gateway-toggle-button',
+		toggleHidden: 'tribe-common-a11y-hidden',
```
Comment (juanfra) [inline src/resources/js/commerce/gateway/toggler.js:75]: Normally, [when we use class names or selectors](https://github.com/the-events-calendar/event-tickets/blob/master/src/resources/js/v2/tickets-block.js#L41) we include the dot `.`. And then we have a prototype function which is `.className()` to get it without the dot.

i.e.: `obj.toggler.classes.toggleHidden.className()`.

I think the best idea would be to have consistency with this and keep doing things like we're doing it on the other libraries. (also the key that it's normally used for the a11y class is `hiddenElement` to make it more abstract and not only related to the toggle.)

File: src/resources/js/commerce/gateway/toggler.js [hunk 6]
```diff
@@ -113,7 +114,10 @@ tribe.tickets.commerce.gateway = tribe.tickets.commerce.gateway || {};
 	 * @param {Element} gateway Gateway element to show.
 	 */
 	obj.toggler.showGateway = gateway => {
```
Comment (juanfra) [inline src/resources/js/commerce/gateway/toggler.js:116]: I'm not entirely sure why we use the `obj.toggler` prefix. We could remove the `toggler` part here, and addit in the obj (the last line, make it `})( jQuery, tribe.tickets.commerce.gateway.toggler );`)

We'll also need to define the toggler beforehand, up in the code: `tribe.tickets.commerce.gateway.toggler = tribe.tickets.commerce.gateway.toggler || {};`

I think we should keep doing that for consistency between libraries.

File: src/resources/js/commerce/gateway/toggler.js [hunk 7]
```diff
@@ -141,7 +148,7 @@ tribe.tickets.commerce.gateway = tribe.tickets.commerce.gateway || {};
 			obj.toggler.hideGateway( gateway );
 		});
 		obj.toggler.toggles.default.addClass(obj.toggler.classes.toggleOpen).hide().attr( 'aria-hidden', 'true' );
-		obj.toggler.toggles.additional.removeClass(obj.toggler.classes.toggleOpen);
+		obj.toggler.toggles.additional.removeClass(obj.toggler.classes.toggleOpen).attr( 'aria-selected', 'false' );
```
Comment (juanfra) [inline src/resources/js/commerce/gateway/toggler.js]: If I'm not mistaken we're applying this to the `containers` of all the additional gateways which are not the default, no?

The aria-selected should be applied to the "opener" and not the container.

File: src/resources/js/commerce/gateway/toggler.js [hunk 8]
```diff
@@ -157,7 +164,7 @@ tribe.tickets.commerce.gateway = tribe.tickets.commerce.gateway || {};
 			}
 			obj.toggler.showGateway( gateway );
 		});
-		obj.toggler.toggles.additional.addClass(obj.toggler.classes.toggleOpen);
+		obj.toggler.toggles.additional.addClass(obj.toggler.classes.toggleOpen).attr( 'aria-selected', 'true' );
```
Comment (juanfra) [inline src/resources/js/commerce/gateway/toggler.js]: If I'm not mistaken we're applying this to the `containers` of all the additional gateways which are not the default, no? I may be reading this the wrong way.  The aria-selected should be applied to the "opener" and not the container.

File: src/resources/js/commerce/gateway/toggler.js (hunk 5 above)
Comment (codingmusician) [reply src/resources/js/commerce/gateway/toggler.js]: Nice! Didn't know about that. Will implement it.

File: src/resources/js/commerce/gateway/toggler.js (hunk 6 above)
Comment (codingmusician) [reply src/resources/js/commerce/gateway/toggler.js]: Good idea. Thanks!

File: src/resources/js/commerce/gateway/toggler.js (hunk 8 above)
Comment (codingmusician) [reply src/resources/js/commerce/gateway/toggler.js]: This is actually for the toggle button itself. The toggle button is the `opener`

File: src/resources/js/commerce/gateway/toggler.js (hunk 7 above)
Comment (codingmusician) [reply src/resources/js/commerce/gateway/toggler.js]: This is actually for the toggle button itself. The toggle button is the `opener`

**Review by bordoni — APPROVED** [review]

**Pull request merged**